tests/components/netatmo/test_init.py

```diff
@@ -0,0 +1,210 @@
+        )
+
+    await hass.async_block_till_done()
+
+    assert mock_auth.call_count == 1
+    assert mock_impl.call_count == 1
+    assert mock_webhook.call_count == 1
+
+    assert config_entry.state == config_entries.ENTRY_STATE_LOADED
+    assert len(hass.config_entries.async_entries(DOMAIN)) == 1
+    assert len(hass.states.async_all()) > 0
+
+
+async def test_setup_component_with_webhook(hass, entry):
+    """Test setup and teardown of the netatmo component with webhook registration."""
+    webhook_id = entry.data[CONF_WEBHOOK_ID]
+    await simulate_webhook(hass, webhook_id, FAKE_WEBHOOK_ACTIVATION)
+    await hass.async_block_till_done()
+
+    assert len(hass.states.async_all()) > 0
+
+    webhook_id = entry.data[CONF_WEBHOOK_ID]
+    await simulate_webhook(hass, webhook_id, FAKE_WEBHOOK_ACTIVATION)
+    await hass.async_block_till_done()
+
+    # Assert webhook is established successfully
+    climate_entity_livingroom = "climate.netatmo_livingroom"
+    assert hass.states.get(climate_entity_livingroom).state == "auto"
+    await simulate_webhook(hass, webhook_id, FAKE_WEBHOOK)
+    assert hass.states.get(climate_entity_livingroom).state == "heat"
+
+    for config_entry in hass.config_entries.async_entries("netatmo"):
+        await hass.config_entries.async_remove(config_entry.entry_id)
+
+    await hass.async_block_till_done()
+    assert len(hass.states.async_all()) == 0
+
+
+async def test_setup_without_https(hass, config_entry):
+    """Test if set up with cloud link and without https."""
+    hass.config.components.add("cloud")
+    with patch(
+        "homeassistant.helpers.network.get_url",
+        return_value="https://example.nabu.casa",
+    ), patch(
+        "homeassistant.components.netatmo.api.ConfigEntryNetatmoAuth"
+    ) as mock_auth, patch(
+        "homeassistant.helpers.config_entry_oauth2_flow.async_get_config_entry_implementation",
+    ), patch(
+        "homeassistant.components.webhook.async_generate_url"
+    ) as mock_webhook:
+        mock_auth.return_value.post_request.side_effect = fake_post_request
+        mock_webhook.return_value = "http://example.com"
+        assert await async_setup_component(
+            hass, "netatmo", {"netatmo": {"client_id": "123", "client_secret": "abc"}}
+        )
+
+    await hass.async_block_till_done()
+
+    webhook_id = config_entry.data[CONF_WEBHOOK_ID]
+    await simulate_webhook(hass, webhook_id, FAKE_WEBHOOK_ACTIVATION)
+    await hass.async_block_till_done()
+
+    # Assert webhook is established successfully
+    climate_entity_livingroom = "climate.netatmo_livingroom"
+    assert hass.states.get(climate_entity_livingroom).state == "auto"
+    await simulate_webhook(hass, webhook_id, FAKE_WEBHOOK)
+    assert hass.states.get(climate_entity_livingroom).state == "auto"
+
+
+async def test_setup_with_cloud(hass, config_entry):
+    """Test if set up with active cloud subscription."""
+    await mock_cloud(hass)
+    hass.data["cloud"].id_token = jwt.encode(
```

Comment:
Patch the cloud integration functions instead:

https://github.com/home-assistant/core/blob/46a3b80a2dbe217d35d906ce7a29c5ae0aa47523/homeassistant/components/cloud/__init__.py#L113

https://github.com/home-assistant/core/blob/46a3b80a2dbe217d35d906ce7a29c5ae0aa47523/homeassistant/components/cloud/__init__.py#L120

https://github.com/home-assistant/core/blob/46a3b80a2dbe217d35d906ce7a29c5ae0aa47523/homeassistant/components/cloud/__init__.py#L126